### PR TITLE
Week - 1

### PR DESCRIPTION
**Pull request by EliaYazdi**

### CODE REVIEW

File: Week1/homework/app.js [hunk 1]
```diff
@@ -1,11 +1,92 @@
...
+    author: 'Louisa May',
+    language: 'English',
+    cover: 'Little.png'
+  }
+]
+
+document.body.onload = printBooks;
+//now I want to print all the information in the DOM.
+
+function createAndAppend(typ, parent, attributes = {}) {
```
Comment (grod220) [inline Week1/homework/app.js:75]: Ah, looks like this helper function makes a surprise appearance from the JS3 homework :)

Comment (grod220) [reply Week1/homework/app.js:75]: Just make sure you know what it is doing though 👍

File: Week1/homework/app.js [hunk 2]
```diff
@@ -1,11 +1,92 @@
...
+function printBooks() {
+  const h1 = createAndAppend('h1', document.body, { innerText: 'My Must Read Books' });
+  const ul = createAndAppend('ul', document.body);
+  for (const book of books) {
+    const li = createAndAppend('li', ul);
+    const h2 = createAndAppend('h2', li, { innerText: book.title });
+    const author = createAndAppend('h4', li, { innerText: `Author: ${book.author}` })
+    const language = createAndAppend('h4', li, { innerText: `Language: ${book.language}` })
+    const img = createAndAppend('img', li, { src: book.cover, height: 350, width: 250 })
+  }
```
Comment (grod220) [inline Week1/homework/app.js:91]: Nice, clean and understandable code 👍

File: Week2/homework/map-filter.js [hunk 3]
```diff
@@ -1,11 +1,17 @@
...
 const myNumbers = [1, 2, 3, 4];
+
+const doubleOddNumbers = numbers => {
+  const odds = numbers.filter(function(number) {
+    return number % 2 !== 0;
+  });
+  const doubles = odds.map(function(number) {
+    return number * 2;
+  });
+  return doubles;
```
Comment (grod220) [inline Week2/homework/map-filter.js:12]: Yep this works. You can even do this by chaining:

```
numbers.filter(number => number % 2 !== 0).map(number => number * 2)
```

File: Week2/homework/squirtle-sprites.js [hunk 4]
```diff
@@ -4,8 +4,32 @@
   Call this function to get a JSON string of the data
   (simulates calling a server to retrieve data)
 */
+function createAndAppend(typ, parent, attributes = {}) {
+  const elem = document.createElement(typ);
+  parent.appendChild(elem);
+  for (const key in attributes) elem[key] = attributes[key]
+  return elem
+}
```
Comment (grod220) [inline Week2/homework/squirtle-sprites.js:12]: Don't think this is used below.

Comment (grod220) [reply Week2/homework/squirtle-sprites.js:12]: Can be removed

File: Week3/homework/step2-1.js [hunk 5]
```diff
@@ -1,9 +1,8 @@
 'use strict';
 
-function foo(func) {
-  // What to do here?
-  // Replace this comment and the next line with your code
-  console.log(func);
+function foo() {
+
+  console.log('Week3');
 }
```
Comment (grod220) [inline Week3/homework/step2-1.js:6]: Hmm. Actually, Foo should accept a function and then invoke/call that function. Don't see that happening here.

File: Week3/homework/step2-2.js [hunk 6]
```diff
@@ -2,22 +2,32 @@
...
+  for (let i = startIndex; i <= stopIndex; i++) {
+    numbers.push(i);
+    if (i % 3 === 0) {
+      threeCallback();
+    }
+    if (i % 5 === 0) {
+      fiveCallback();
+    }
+    if (i % 3 === 0 && i % 5 === 0) {
+      threeCallback(fiveCallback());
```
Comment (grod220) [inline Week3/homework/step2-2.js:14]: They shouldn't be called within each other, but separately

File: Week3/homework/step2-2.js [hunk 7]
```diff
@@ -2,22 +2,32 @@
...
-
-  // Replace this comment and the next line with your code
-  console.log(startIndex, stopIndex, threeCallback, fiveCallback, numbers);
+  for (let i = startIndex; i <= stopIndex; i++) {
+    numbers.push(i);
+    if (i % 3 === 0) {
+      threeCallback();
+    }
+    if (i % 5 === 0) {
+      fiveCallback();
```
Comment (grod220) [inline Week3/homework/step2-2.js:11]: fiveCallback & threeCallback take a number. So you should pass in the index into them.

File: Week3/homework/step2-3.js [hunk 8]
```diff
@@ -5,21 +5,27 @@ function repeatStringNumTimesWithFor(str, num) {
   // eslint-disable-next-line prefer-const
```
Comment (grod220) [inline Week3/homework/step2-3.js:5]: Please remove all eslint-disable comments

File: Week3/homework/step2-3.js [hunk 9]
```diff
@@ -5,21 +5,27 @@ function repeatStringNumTimesWithFor(str, num) {
...
   let result = '';
 
-  // Replace this comment and the next line with your code
-  console.log(str, num, result);
+  for (let i = 0; i < num; i++) {
+    result = +str;
```
Comment (grod220) [inline Week3/homework/step2-3.js:9]: Here is the bug. The correct shorthand should be:

`result += str`
or 
`result = result + str`

File: Week3/homework/step2-3.js [hunk 10]
```diff
@@ -30,9 +36,12 @@ console.log('while', repeatStringNumTimesWithWhile('abc', 3));
 function repeatStringNumTimesWithDoWhile(str, num) {
   // eslint-disable-next-line prefer-const
   let result = '';
+  let i = 0;
 
-  // Replace this comment and the next line with your code
-  console.log(str, num, result);
+  do {
+    result = +str;
```
Comment (grod220) [inline Week3/homework/step2-3.js:42]: Same bug 🐛 here

**Review by grod220 — CHANGES_REQUESTED** [review]
Do you have the remaining week3 questions as well?

File: Week1/homework/app.js (hunk 1 above)
Comment (EliaYazdi) [reply Week1/homework/app.js:75]: Yeah , Actually I did it in a different way , but in the session with Christer we changed it to look like this. I think I got how it works. :)

File: Week1/homework/app.js (hunk 2 above)
Comment (EliaYazdi) [reply Week1/homework/app.js:91]: Thanks a lot :)

File: Week2/homework/squirtle-sprites.js (hunk 4 above)
Comment (EliaYazdi) [reply Week2/homework/squirtle-sprites.js:12]: you'r right. Done that!

File: Week3/homework/step2-1.js (hunk 5 above)
Comment (EliaYazdi) [reply Week3/homework/step2-1.js:6]: @grod220 I changed it.

File: Week3/homework/step2-2.js (hunk 6 above)
Comment (EliaYazdi) [reply Week3/homework/step2-2.js:14]: @grod220 I called them outside the forloop but inside the function.

Comment (grod220) [reply Week3/homework/step2-2.js:14]: Not sure what you mean. They should look like:
```
threeCallback(i)
fiveCallback(i)
```
See what I mean?

Comment (EliaYazdi) [reply Week3/homework/step2-2.js:14]: function threeFive(startIndex, stopIndex, threeCallback, fiveCallback) {
  const numbers = [];
  for (let i = startIndex; i <= stopIndex; i++) {
    numbers.push(i);
    if (i % 3 === 0) {

    }
    if (i % 5 === 0) {

    }
    if (i % 3 === 0 && i % 5 === 0) {

    }
  }
  threeCallback(fiveCallback());
  fiveCallback(i);
  threeCallback(i);
}

like this?

**Pull request merged**